prediction_market_agent/db/pinecone_handler.py

```diff
@@ -109,15 +111,37 @@ def deduplicate_markets(markets: list[OmenMarket]) -> list[OmenMarket]:
 
         return list(unique_market_titles.values())
 
-    def insert_all_omen_markets_if_not_exists(
-        self, created_after: DatetimeUTC | None = None
-    ) -> None:
+    def update_markets(self, clean_resolved_markets=True) -> None:
+        """We use the agent's run to add embeddings of new markets that don't exist yet in the
+        vector DB."""
+        self.insert_all_omen_markets_if_not_exists()
+        if clean_resolved_markets:
+            self.remove_resolved_markets()
+
+    def remove_resolved_markets(self) -> None:
+        """Remove markets that are closed but still present in the index."""
+        existing_ids = self.get_existing_ids_in_index()
+        resolved_markets = OmenSubgraphHandler().get_omen_binary_markets(
+            limit=sys.maxsize,
+            resolved=True,
+        )
+        resolved_markets_id = [
+            self.encode_text(m.question_title)
+            for m in resolved_markets
+            if m.is_resolved
+        ]
+        ids_to_delete = set(existing_ids).intersection(resolved_markets_id)
+        self.remove_markets_by_id(ids_to_delete=list(ids_to_delete))
```

Comment:
Pinecone is used by both ThinkThoroughly and Arbitrage Agent. Closed markets are useful for ThinkThoroughly but unusable for Arbitrage. 

Won't this delete the closed markets from the database itself? E.g. also for TT agent?

Reply:
As per our huddle, removed deletion of markets, might be useful to TTA. Closing this.